src/hooks/useQueryString.js

```diff
@@ -43,10 +43,13 @@ function useQueryString(queryKey, resetValue) {
     }
     queryStrings[queryKey] = queryValue;
     const updatedQs = queryString.stringify(queryStrings);
-    history.push({
-      pathname,
-      search: queryValue === resetValue ? null : updatedQs,
-    });
+    // Don't allow empty query string values
+    if (queryValue) {
```

Comment:
I'm curious to find out why this change was necessary? Technically a query parameter without a value is still valid so I don't think this is the correct fix for whatever issue it's resolving.

Reply:
It was because when closing the panel, the url would be `/model?panel=&app=`

This may however be a symptom of the other issue you've described.